src/main/java/com/google/sps/data/Section.java

```diff
@@ -15,33 +15,58 @@
 package com.google.sps.data;
 
 import java.util.Collections;
+import java.util.LinkedList;
 import java.util.List;
 
 public class Section {
-    private String professor;
-    // Meeting times will be sorted in ascending order of time, 
-    // with the first lesson being first in the list
-    private List<TimeRange> meetingTimes;
-
-    public Section(String professor, List<TimeRange> meetingTimes){
-        this.professor = professor;
-        this.meetingTimes = meetingTimes;
-        Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
-
-        for(int i = 0; i < meetingTimes.size() - 1; i++){
-          if(meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
-            throw new IllegalArgumentException("A section's lecture times can't overlap!");
-          }
-        }
+  private String professor;
+  // Meeting times will be sorted in ascending order of time,
+  // with the first lesson being first in the list
+  private List<TimeRange> meetingTimes;
+
+  public Section(String professor, List<TimeRange> meetingTimes) {
+    this.professor = professor;
+    this.meetingTimes = meetingTimes;
+    Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
+
+    for (int i = 0; i < meetingTimes.size() - 1; i++) {
+      if (meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
+        throw new IllegalArgumentException("A section's lecture times can't overlap!");
+      }
     }
+  }
+
+  public List<TimeRange> getMeetingTimes() {
+    return this.meetingTimes;
+  }
+
+  /**
+   * Function to return whether or not a section overlaps with another.
+   * 
+   * @param other The other section.
+   * @return If there's overlap between the current section and the other section
+   */
+  public boolean overlaps(Section other) {
+    LinkedList<TimeRange> thisTimes = new LinkedList<>(this.meetingTimes);
+    LinkedList<TimeRange> otherTimes = new LinkedList<>(other.getMeetingTimes());
 
-    /**
-     * Function to return whether or not a section overlaps with another
-     * @param other
-     * @return If there's overlap between the current section and the other section
-     */
-    public boolean overlaps(Section other){
-        //TODO: This
+    TimeRange thisCurr = thisTimes.removeFirst();
+    TimeRange otherCurr = otherTimes.removeFirst();
+
+    while (!thisTimes.isEmpty() && !otherTimes.isEmpty()) {
+      if (thisCurr.overlaps(otherCurr)) {
         return false;
+      } else if (thisCurr.end() > otherCurr.end()) {
```

Comment:
I might be misunderstanding, but if thisCurr ends later than otherCurr, and the sections are sorted in ascending order, shouldn't you move on to the next (later) section in otherTimes, not thisTimes? (ie if thisCurr is 10-11, and otherCurr is 8-9 (so 11 > 9), if you move onto a later thisTimes section, they will never overlap?) Sorry if I didn't follow the code right!

Reply:
You're 100% correct, good catch.